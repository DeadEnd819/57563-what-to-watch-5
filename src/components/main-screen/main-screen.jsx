import React, {Fragment} from "react";
import {connect} from "react-redux";
import PropTypes from "prop-types";
import Header from "../header/header";
import Footer from "../footer/footer";
import MovieMenu from "../movie-menu/movie-menu";
import MovieCatalog from "../movie-catalog/movie-catalog";
import {PromoTypes} from "../../prop-types/prop-types";
import {isUserLoggedIn} from "../../store/selectors";

const MainScreen = ({promoFilm, isUserLogged}) => {
  const {name, backgroundImage, posterImage} = promoFilm;

  return <Fragment>
    <section className="movie-card">
      <div className="movie-card__bg">
        <img src={backgroundImage} alt={name}/>
      </div>

      <h1 className="visually-hidden">WTW</h1>

      <Header className={`movie-card__head`}/>

      <div className="movie-card__wrap">
        <div className="movie-card__info">
          <div className="movie-card__poster">
            <img src={posterImage} alt={`${name} poster`} width="218" height="327"/>
          </div>

          <MovieMenu film={promoFilm} isMain={true} isUserLogged={isUserLogged}/>
        </div>
      </div>
    </section>

    <div className="page-content">

      <MovieCatalog />

      <Footer />

    </div>
  </Fragment>;
};

MainScreen.propTypes = {
  promoFilm: PromoTypes.isRequired,
  isUserLogged: PropTypes.bool.isRequired,
};

const mapStateToProps = (state) => ({
  promoFilm: state.DATA.promo,
  isUserLogged: isUserLoggedIn(state),
});

export {MainScreen};
export default connect(mapStateToProps)(MainScreen);

