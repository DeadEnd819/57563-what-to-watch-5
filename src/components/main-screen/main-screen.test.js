import React from "react";
import renderer from "react-test-renderer";
import {MainScreen} from "./main-screen";

const promoFilm = {
  id: 1,
  name: `The Grand Budapest Hotel`,
  posterImage: `the-grand-budapest-hotel-poster.jpg`,
  backgroundImage: `bg-the-grand-budapest-hotel.jpg`,
  backgroundColor: `#000000`,
  genre: `Drama`,
  released: 2014,
  isFavorite: false,
};

jest.mock(`../header/header.jsx`, () => `Header`);
jest.mock(`../footer/footer.jsx`, () => `Footer`);
jest.mock(`../movie-menu/movie-menu.jsx`, () => `MovieMenu`);
jest.mock(`../movie-catalog/movie-catalog.jsx`, () => `MovieCatalog`);

const noop = ()=>{};

it(`Should MainScreen render correctly`, () => {
  const tree = renderer
    .create(<MainScreen
      promoFilm={promoFilm}
      resetFilterAction={noop}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
