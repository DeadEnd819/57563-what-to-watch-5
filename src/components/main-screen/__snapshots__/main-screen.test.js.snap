// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should MainScreen render correctly 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="The Grand Budapest Hotel"
        src="bg-the-grand-budapest-hotel.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <Header
      className="movie-card__head"
    />
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="The Grand Budapest Hotel poster"
            height="327"
            src="the-grand-budapest-hotel-poster.jpg"
            width="218"
          />
        </div>
        <MovieMenu
          film={
            Object {
              "backgroundColor": "#000000",
              "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
              "genre": "Drama",
              "id": 1,
              "isFavorite": false,
              "name": "The Grand Budapest Hotel",
              "posterImage": "the-grand-budapest-hotel-poster.jpg",
              "released": 2014,
            }
          }
        />
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <MovieCatalog />
    <Footer />
  </div>,
]
`;
